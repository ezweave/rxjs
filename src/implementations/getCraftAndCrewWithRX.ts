import axios from 'axios';
import { transformAllFieldNamesToCamel } from 'field-caser';
import { map, mergeAll, mergeMap, zip } from 'rxjs';

import { getNPeoplePromises } from '@rx-example/services';
import { getStarship } from '@rx-example/services/getStarship';
import { People, Starship } from '@rx-example/types';

export const getCraftAndCrewRX = (n, httpGet = axios.get) =>
  new Promise((resolve) =>
    zip(...getNPeoplePromises(httpGet)(n))
      .pipe(
        map((person) => <People>transformAllFieldNamesToCamel(person)),
        map(getStarship(httpGet)),
        mergeAll(),
        mergeMap(async (starshipPromise) => {
          const starship = await starshipPromise;
          return <Starship>transformAllFieldNamesToCamel(starship);
        }),
      )
      .subscribe(resolve),
  );
