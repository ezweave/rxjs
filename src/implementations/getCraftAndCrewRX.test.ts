import { getCraftAndCrewRX } from './getCraftAndCrewWithRX';

describe(getCraftAndCrewRX, () => {
  it('gets nnn', async () => {
    const stuff = await getCraftAndCrewRX(2);
    expect(stuff).toMatchSnapshot();
  });
});
