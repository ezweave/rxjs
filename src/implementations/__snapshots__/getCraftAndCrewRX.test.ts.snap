// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCraftAndCrewRX gets nnn 1`] = `
SafeSubscriber {
  "_finalizers": Array [
    OperatorSubscriber {
      "_complete": [Function],
      "_error": [Function],
      "_finalizers": Array [
        OperatorSubscriber {
          "_complete": [Function],
          "_error": [Function],
          "_finalizers": Array [
            [Function],
            OperatorSubscriber {
              "_complete": [Function],
              "_error": [Function],
              "_finalizers": null,
              "_next": [Function],
              "_parentage": [Circular],
              "closed": false,
              "destination": [Circular],
              "initialTeardown": undefined,
              "isStopped": false,
              "onFinalize": undefined,
              "shouldUnsubscribe": undefined,
            },
            [Function],
          ],
          "_next": [Function],
          "_parentage": [Circular],
          "closed": false,
          "destination": [Circular],
          "initialTeardown": undefined,
          "isStopped": false,
          "onFinalize": undefined,
          "shouldUnsubscribe": undefined,
        },
      ],
      "_next": [Function],
      "_parentage": [Circular],
      "closed": false,
      "destination": [Circular],
      "initialTeardown": undefined,
      "isStopped": false,
      "onFinalize": undefined,
      "shouldUnsubscribe": undefined,
    },
  ],
  "_parentage": null,
  "closed": false,
  "destination": ConsumerObserver {
    "partialObserver": Object {
      "complete": undefined,
      "error": undefined,
      "next": [Function],
    },
  },
  "initialTeardown": undefined,
  "isStopped": false,
}
`;
